test/uk/gov/hmrc/gform/sharedmodel/formtemplate/BooleanExprSpec.scala

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.sharedmodel.formtemplate
+
+import org.scalatest.mockito.MockitoSugar.mock
+import uk.gov.hmrc.gform.Spec
+import uk.gov.hmrc.gform.auth.models.Retrievals
+import uk.gov.hmrc.gform.sharedmodel.ExampleData
+import uk.gov.hmrc.gform.sharedmodel.form.FormField
+
+class BooleanExprSpec extends Spec {
```

Comment:
There is huge amount of repetition in this Spec. I deem repetition being of secondary concern though, my primary concern is that it is hard to see what is being tested. Given all of this it seems like this spec is great candidate for TableDriven test approach.

Here is [example of it](https://github.com/hmrc/gform/blob/master/test/uk/gov/hmrc/gform/sharedmodel/JsonParseTestFormat.scala#L82-L93) in gform